src/main/scala/com/microsoft/ml/spark/vw/featurizer/NumericFeaturizer.scala

```diff
@@ -14,35 +14,47 @@ import scala.collection.mutable
   * @param columnName used as feature name prefix.
   * @param namespaceHash pre-hashed namespace.
   * @param mask bit mask applied to final hash.
-  * @param getFieldValue lambda to unify the cast/conversion to double.
   */
-class NumericFeaturizer(override val fieldIdx: Int, override val columnName: String, namespaceHash: Int,
-                        mask: Int, val getFieldValue: Row => Double)
-  extends Featurizer(fieldIdx) {
+class NumericFeaturizer[T <: AnyVal{ def toDouble:Double }](override val fieldIdx: Int,
+                                 override val columnName: String,
+                                 val namespaceHash: Int,
+                                 val mask: Int,
+                                 val zero: T)
+  extends Featurizer(fieldIdx) with ElementFeaturizer[T] {
 
   /**
     * Pre-hashed feature index.
     */
   val featureIdx = mask & VowpalWabbitMurmur.hash(columnName, namespaceHash)
 
-  /**
-    * Featurize a single row.
-    * @param row input row.
-    * @param indices output indices.
-    * @param values output values.
-    * @note this interface isn't very Scala-esce, but it avoids lots of allocation.
-    *       Also due to SparseVector limitations we don't support 64bit indices (e.g. indices are signed 32bit ints)
-    */
   override def featurize(row: Row,
                          indices: mutable.ArrayBuilder[Int],
                          values: mutable.ArrayBuilder[Double]): Unit = {
-    val value = getFieldValue(row)
+      featurize(0, row.getAs[T](fieldIdx), indices, values)
+  }
 
+  def featurize(idx: Int,
+                value: T,
+                indices: mutable.ArrayBuilder[Int],
+                values: mutable.ArrayBuilder[Double]): Unit = {
     // Note: 0 valued features are always filtered.
-    if (value != 0) {
-        indices += featureIdx
-        values += value
+    if (value != zero) {

```

Comment:
any reason why this wont be 0?

Reply:
If it is zero it isn't used by the learner and can be filtered out